dashpay/src/main/kotlin/org/dashevo/dashpay/BlockchainIdentity.kt

```diff
@@ -1023,4 +1050,129 @@ class BlockchainIdentity {
         }
         return null
     }
+
+    // Contact Requests
+    fun addContactToWallet(contactKeyChain: FriendKeyChain) {
+        when (contactKeyChain.type) {
+            FriendKeyChain.KeyChainType.RECEIVING_CHAIN -> wallet!!.addReceivingFromFriendKeyChain(contactKeyChain)
+            FriendKeyChain.KeyChainType.SENDING_CHAIN -> wallet!!.addSendingToFriendKeyChain(contactKeyChain)
+        }
+        //TODO: Save the wallet here
```

Comment:
Will we do it later, any reason we can't do it now?

Reply:
One reason is because the `Wallet` object doesn't know what filename to save it to.  Neither does `BlockchainIdentity` Currently, this saving the wallet file is handled by the `WalletApplication` class in the app.  

Perhaps we could add a listener that is triggered when KeyChains are added to the wallet, then the listener (app in our case) will perform the saving.

There is no listener currently for adding a key chain.  Previously, such a think only happens when the app is installed or a wallet is restored.

Reply:
Actually I was wrong.  The wallet object keeps track of the filename.  Changes were made here to ensure that the wallet file is saved. 

https://github.com/dashevo/dashj/pull/72/commits/9ad8273754dbceead0245d9ea41fcf8d70f43d88

We won't need to do it here, I will remove the comment about saving.

Reply:
@sambarboza good catch.  This gets rid of one future bug.